function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautAwsProxyRequest.java

```diff
@@ -475,12 +506,24 @@ public <T> Optional<T> get(CharSequence name, ArgumentConversionContext<T> conve
      */
     private class AwsHeaders implements HttpHeaders {
 
-        private Headers multiValueHeaders = awsProxyRequest.getMultiValueHeaders();
+        private final Headers multiValueHeaders = awsProxyRequest.getMultiValueHeaders();
```

Comment:
I think we could resolve the complexity of the class by resolving the headers in the constructor.